aten/src/ATen/native/mkldnn/TensorFactories.cpp

```diff
@@ -4,23 +4,23 @@ namespace at { namespace native {
 
 #if AT_MKLDNN_ENABLED()
 
-Tensor empty_mkldnn(IntArrayRef sizes, const TensorOptions& options, c10::optional<c10::MemoryFormat> optional_memory_format) {
-  TORCH_CHECK(
-     !options.has_memory_format(),
-     "'memory_format' argument is incompatible with mkldnn tensor");
+Tensor empty_mkldnn(IntArrayRef sizes, c10::optional<ScalarType> dtype, c10::optional<Layout> layout, c10::optional<Device> device, c10::optional<bool> pin_memory, c10::optional<c10::MemoryFormat> optional_memory_format) {
   TORCH_CHECK(
      !optional_memory_format.has_value(),
      "'memory_format' argument is incompatible with mkldnn tensor");
+  TORCH_CHECK(
+    !layout.has_value(),
+    "'layout' argument is incompatible with mkldnn tensor");
```

Comment:
New!

Reply:
yes but I thought adding it makes sense. Do you think we should not add it?

Reply:
ok I removed it